target/linux/rtl838x/image/Makefile

```diff
@@ -0,0 +1,107 @@
+#
+# Copyright (C) 2008-2011 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/image.mk
+
+KERNEL_LOADADDR = 0x80000000
+KERNEL_ENTRY = 0x80000400
+
+define Device/Default
+  PROFILES = Default
+  KERNEL := kernel-bin | append-dtb | gzip | uImage gzip
+  KERNEL_INITRAMFS := kernel-bin | append-dtb | gzip | uImage gzip
+  DEVICE_DTS_DIR := ../dts
+  DEVICE_DTS = $$(SOC)_$(1)
+  IMAGES := sysupgrade.bin
+  SUPPORTED_DEVICES := $(subst _,$(comma),$(1))
+  sysupgrade_bin := append-kernel | pad-to 64k | append-rootfs | pad-rootfs
+  IMAGE/sysupgrade.bin := append-kernel | pad-to 64k | append-rootfs | pad-rootfs | append-metadata | check-size $$$$(IMAGE_SIZE)
+endef
+
+define Build/netgear-uImage
+	$(call Build/uImage,$(1) -M $(NETGEAR_KERNEL_MAGIC))
+endef
+
+define Build/custom-uimage
+	mkimage -A $(LINUX_KARCH) \
+		-O linux -T kernel \
+		-C gzip -a $(KERNEL_LOADADDR) $(if $(UIMAGE_MAGIC),-M $(UIMAGE_MAGIC),) \
+		-e $(if $(KERNEL_ENTRY),$(KERNEL_ENTRY),$(KERNEL_LOADADDR)) \
+		-n '$(1)' -d $@ $@.new
+	mv $@.new $@
+endef
+
+define Device/allnet_all-sg8208m
+  SOC := rtl8382
+  IMAGE_SIZE := 7168k
+  DEVICE_VENDOR := ALLNET
+  DEVICE_MODEL := ALL-SG8208M
+  UIMAGE_MAGIC := 0x00000006
+  KERNEL := kernel-bin | append-dtb | gzip | custom-uimage 2.2.2.0
+  KERNEL_INITRAMFS := kernel-bin | append-dtb | gzip | custom-uimage 2.2.2.0
+  DEVICE_PACKAGES := ip-full ip-bridge kmod-gpio-button-hotplug tc
```

Comment:
I don't think ip-full and tc will be acceptable for default setup.

Reply:
These switches have a full DSA interface, but they cannot be configured with swconfig. So tools from ip-bridge are necessary. The mirroring functionality needs tc to be supported. The Flash memories of the devices are relatively large for NOR flashes, always at least 16MB, normally 32. So it does not hurt to provide the functionality to be able to setup the switch...